File: communicationAPI/src/main/java/eu/cybergeiger/communication/Declaration.java
package eu.cybergeiger.communication;

/**
 * <p>The self declaration of the plugin.</p>
 *
 * <p>
 * If a plugin declares data sharing only access to own data is granted. If a plugin declares no sharing full access
 * to all available data is granted.
 * </p>
 */
public enum Declaration {
  DO_NOT_SHARE_DATA("This plugin does not share any device, company, or user related "
      + "data with or without consent to any party within or outside this device."),
  DO_SHARE_DATA("This plugin does share device, company, or user related "
      + "data with consent a other apps or parties within or outside this device.");

  private String declaration;

  Declaration(String declaration) {
    this.declaration = declaration;
  }

  public String getDeclaration() {
    return this.declaration;
  }
}

File: lib/src/plugin/declaration.dart
library geiger_api;

/// The self declaration of the plugin.
enum Declaration {
  /// No data sharing is done and thus no implications on GDPR
  doNotShareData,

  /// Data sharing is done. As a result only access to own objects (any TLP) is granted and to all
  /// objects of TLP:WHITE
  doShareData
}

extension DeclarationExtension on Declaration {
  String getDeclaration() {
    switch (this) {
      case Declaration.doNotShareData:
        return 'This plugin does not share any device, company, or user related '
            'data with or without consent to any party within or outside this device.';
      case Declaration.doShareData:
        return 'This plugin does share device, company, or user related '
            'data with consent a other apps or parties within or outside this device.';
    }
  }
}
